### PR TITLE
fix: address codeql warning with hostname matches

### PR DESCRIPTION
**Pull request by JustinBeckwith**
I never write a regex without @alexander-fenster checking me first ...

### CODE REVIEW

**Review by bcoe — APPROVED** [review]

File: src/gaxios.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ function skipProxy(url: string) {
   const parsedURL = new URL(url);
   return !!noProxyUrls.find(url => {
     if (url.startsWith('*.') || url.startsWith('.')) {
-      url = url.replace('*', '');
+      url = url.replace(/^\*\./, '.');
```
Comment (bcoe) [inline src/gaxios.ts:86]: This behavior looks a little different, I think we were originally replacing a leading `*` with ``.

Now we replace a leading `*.` with ``.

I believe these are the two unit tests that this behavior is meant to facilitate:

```js
    it('should proxy if no_proxy env variable has asterisk, but URL is not matching', async () => {
      const url = 'https://domain.example2.com';
      sandbox.stub(process, 'env').value({
        https_proxy: 'https://fake.proxy',
        no_proxy: '*.example.com',
      });
      const body = {hello: '🌎'};
      const scope = nock(url).get('/').reply(200, body);
      const res = await request({url});
      scope.done();
      assert.deepStrictEqual(res.data, body);
      assert.ok(res.config.agent instanceof HttpsProxyAgent);
    });

    it('should not proxy if no_proxy env variable starts with a dot, and URL partially matches', async () => {
      const url = 'https://domain.example.com';
      sandbox.stub(process, 'env').value({
        https_proxy: 'https://fake.proxy',
        no_proxy: '.example.com',
      });
      const body = {hello: '🌎'};
      const scope = nock(url).get('/').reply(200, body);
      const res = await request({url});
      scope.done();
      assert.deepStrictEqual(res.data, body);
      assert.strictEqual(res.config.agent, undefined);
    });
```

I'm scratching my head and wondering why the second continues working?

**Review by bcoe — CHANGES_REQUESTED** [review]

File: src/gaxios.ts (hunk 1 above)
Comment (JustinBeckwith) [reply src/gaxios.ts:86]: The replace passes in a `'.'` now.

Comment (bcoe) [reply src/gaxios.ts:86]: Can't we get rid of the `|| url.startsWith('.')` then? or what am I missing?

**Review by bcoe — APPROVED** [review]

File: src/gaxios.ts (hunk 1 above)
Comment (JustinBeckwith) [reply src/gaxios.ts:86]: I was trying to change as little as possible 😆  I don't fully understand what this function does.

Comment (bcoe) [reply src/gaxios.ts:86]: The goal is that if you provide a wild card URL like `*.google.com` in NO_PROXY, `foo.google.com`, `bar.google.com`, etc. should all not be proxied. This is also true of providing `.google.com`.

**Pull request merged**